aesara/graph/op.py

```diff
@@ -197,6 +197,10 @@ class Op(MetaObject):
 
     """
 
+    itypes = None
+    otypes = None
+    params_type: Optional[ParamsType] = None
+
```

Comment:
This is not an exclusively type-information-based change; it could actually affect code that depends on `Op` (e.g. similar to how the `hasattr` checks below needed to be updated).

These kinds of changes need their own commits, if only to make the them stand out in some way.

Reply:
I did that already, or do you want them to be split up even further?
![grafik](https://user-images.githubusercontent.com/5894642/154640705-e5c74eba-b3e1-4a11-9313-91938f8e9c13.png)

Reply:
> I did that already, or do you want them to be split up even further?

Yes, each such independent change that affects a different class/area of code is eligible for its own commit, especially when each requires distinct considerations.

"Rectify typing with runtime code changes" is a catch-all that could encompass literally any and all substantial code changes, and such a themed commit requires reviewers to manually organize all the disparate changes before understanding what was really changed and (potentially) why. 

In contrast, other common catch-alls like "Refactor ..." are generally constrained to specific areas of the codebase and/or are assumed to contain no interface or functionality changes.  Those assumptions are much easier to check via a quick pass over the diffs.  

This commit doesn't have the above properties, which makes for a slower review cycle.